cdap-docs/cdap-apps/source/hydrator/creating.rst

```diff
@@ -193,7 +193,275 @@ property replaces the ``schedule`` property of an ETL Batch Application.
+        },
+        "sinks": [
+          {
+            "name": "replicaTable",
+            "plugin": {
+              "name": "Table",
+              "properties": {
+                "schema": "{
+                  \"type\":\"record\",
+                  \"name\":\"etlSchemaBody\",
+                  \"fields\":[
+                    {\"name\":\"userId\",\"type\":\"string\"},
+                    {\"name\":\"purchaseItem\",\"type\":\"string\"},
+                    {\"name\":\"purchasePrice\",\"type\":\"long\"}
+                  ]
+                }",
+                "name": "replicaTable",
+                "schema.row.field": "userId"
+              }
+            }
+          },
+          {
+            "name": "usersTable",
+            "plugin": {
+              "name": "Table",
+              "properties": {
+                "schema": "{
+                  \"type\":\"record\",
+                  \"name\":\"etlSchemaBody\",
+                  \"fields\":[
+                    {\"name\":\"userId\",\"type\":\"string\"}
+                  ]
+                }",
+                "name": "targetCustomers",
+                "schema.row.field": "userId"
+              }
+            }
+          }
+        ],
+        "transforms": [
+          {
+            "name": "spendingUsersScript",
+            "plugin": {
+              "name": "Script",
+              "properties": {
+                "script": "function transform(input, context) {
+                            if (input.purchasePrice > 1000) {
+                              return {'userId' : input.userId};
+                            }
+                          }",
+                "schema": "{
+                  \"type\":\"record\",
+                  \"name\":\"etlSchemaBody\",
+                  \"fields\":[
+                    {\"name\":\"userId\",\"type\":\"string\"}
+                  ]
+                }"
+              }
+            }
+          }
+        ],
+        "connections": [
+          {
+            "from": "purchaseStats",
+            "to": "replicaTable"
+          },
+          {
+            "from": "purchaseStats",
+            "to": "spendingUsersScript"
+          },
+          {
+            "from": "spendingUsersScript",
+            "to": "usersTable"
+          }
+        ],
+        "schedule": "* * * * *",
+        "engine": "mapreduce"
+      }
+    }
+
+**Note:**
+Pipeline connections can be configured to fork from a stage, with the output of the stage sent to
+two or more configured stages; in the above example, the same output record is sent to the other stages.
+Similarly, forked transform stages can merge together at a transform or a sink stage.
+The output schema's from these merging stages must match before joining. Note that
+the order of records sent from the forked stages to the joining stage will not be defined.
+
+In this next example, ``purchaseStream`` has purchase data with fields ``userid``, ``item``, ``count``, and ``price``.
```

Comment:
so this example is meant to illustrate behavior when merging, and the previous example is to illustrate forks?  I think it may make sense to have another section here, and name the previous section 'Forks' or something to that effect.

Otherwise I start reading this and I'm not sure why I need to look at another example.